bindings/netstandard/ElectionGuard/ElectionGuard.Encryption/ExceptionHandler.cs

```diff
@@ -0,0 +1,16 @@
+﻿namespace ElectionGuard
+{
+    public static class ExceptionHandler
+    {
+        public static void GetData(out string function, out string message, out ulong code)
+        {
+            var status = NativeInterface.ExceptionHandler.GetData(out var functionData, out var messageData, out var codeData);
+            function = functionData.PtrToStringUTF8();
+            message = messageData.PtrToStringUTF8();
+            code = codeData;
+            _ = NativeInterface.Memory.FreeIntPtr(functionData);
+            _ = NativeInterface.Memory.FreeIntPtr(messageData);

```

Comment:
similar ot the suggestion below about adding an AsyncSemaphore on the c++ side, we should probably also put some thread safety around freeing these pointers. my concern is that a series of cascading failures across threads on both the managed and unmanaged sides could lead to a deadlock

Reply:
the call to get the strings is making dynamic copies right now so it will not have an issue with the original values.  However we should make sure to add the AsyncSemafore code to the functions that get the strings back as well.  The copy that the C# has is unique to the call it made so we should be ok here with the free calls.